clients/src/main/java/org/apache/kafka/common/requests/InitPidRequest.java

```diff
@@ -23,44 +23,50 @@
 
 public class InitPidRequest extends AbstractRequest {
     private static final String TRANSACTIONAL_ID_KEY_NAME = "transactional_id";
+    private static final String TIMEOUT_KEY_NAME = "timeout";
 
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.INIT_PRODUCER_ID.id);
 
     private final String transactionalId;
+    private final int transactionTimeoutMs;
 
     public static class Builder extends AbstractRequest.Builder<InitPidRequest> {
         private final String transactionalId;
+        private final int transactionTimeoutMs;
 
-        public Builder(String transactionalId) {
+        public Builder(String transactionalId, int transactionTimeoutMs) {
             super(ApiKeys.API_VERSIONS);
             this.transactionalId = transactionalId;
+            this.transactionTimeoutMs = transactionTimeoutMs;
```

Comment:
Deleted by accident: left by @hachikuji to "check if the timeout value is positive".